File: Newishfs/Todos.fs
[<RequireQualifiedAccess>]
module Newishfs.Todos

open System
open Microsoft.AspNetCore.Builder

type Todo =
    { id: int
      title: string
      isDone: bool }

let mutable private todos = []

let getTodos =
    let getTodos () = todos |> List.toSeq
    Func<Todo seq>(getTodos)

let addTodo =
    let addTodo (todo: Todo) =
        todos <- todo :: todos
        true

    Func<Todo, bool>(addTodo)


let registerRoutes (app: WebApplication) =
    app.MapGet("/todos", getTodos) |> ignore
    app.MapPost("/todos", addTodo) |> ignore
    app

File: Newish/Todos.cs
using System;
using System.Collections.Generic;
using Microsoft.AspNetCore.Builder;

namespace Newish
{
    public record Todo(int id, string title, bool isDone);
    public static class Todos
    {
        /// obviously this is not how you should do this, but it's a demo
        private static List<Todo> _todos = new List<Todo>();
        private static Func<IEnumerable<Todo>> GetTodos =
            () => _todos;
        private static Func<Todo, bool> AddTodo =
            todo => { _todos.Add(todo); return true; };

        public static WebApplication RegisterTodoRoutes(this WebApplication app)
        {
            app.MapGet("/todos", GetTodos);
            app.MapPost("/todos", AddTodo);
            return app;
        }

    }
}
